src/main/java/junitparams/internal/parameters/ParametersFromCustomProvider.java

```diff
@@ -28,11 +31,28 @@ public boolean isApplicable() {
     }
 
     private ParametersProvider instantiate(Class<? extends ParametersProvider> providerClass) {
-        try {
-            return providerClass.newInstance();
-        } catch (Exception e) {
-            throw new RuntimeException("Your Provider class must have a public no-arg constructor!", e);
+        Constructor<?>[] constructors = providerClass.getConstructors();
+        for (Constructor<?> constructor : constructors) {
```

Comment:
which is the issue

```java
+            if(parameterTypes.length==1){
...
 +            }else {
 +                try {
 +                    return (ParametersProvider) constructor.newInstance();
 +                } catch (Exception e) {
 +                    throw new RuntimeException("Your Provider class must have a public no-arg constructor!", e);
 +                }
 +            }
```

i.e if length is not 1 then the constructor is treated as a no-args constructor (even for constructor argument lengths > 1)